src/utils.jl

```diff
@@ -207,3 +207,13 @@ function tobuffer(data; kwargs...)
     seekstart(io)
     return io
 end
+
+toidict(x::Base.ImmutableDict) = x
+
+function toidict(pairs)
+    dict = Base.ImmutableDict(first(pairs))
+    for pair in Iterators.drop(pairs, 1)
+        dict = Base.ImmutableDict(dict, pair)
+    end
+    return dict
+end
```

Comment:
```suggestion
toiddict(pairs) = foldl(Base.ImmutableDict, pairs)
```

Reply:
actually had something like this originally but seemed like `foldl` was a bit slower than the more manual loop (didn't investigate why):

```jl
julia> pairs = [randstring(30) => randstring(30) for _ in 1:15];

julia> f(x) = foldl(Base.ImmutableDict, x)
f (generic function with 1 method)

julia> @benchmark f($pairs)
BenchmarkTools.Trial: 10000 samples with 10 evaluations.
 Range (min … max):  1.442 μs … 179.375 μs  ┊ GC (min … max): 0.00% … 98.91%
 Time  (median):     1.479 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):   1.541 μs ±   3.071 μs  ┊ GC (mean ± σ):  3.44% ±  1.71%

       ▃ ▇ █ ▅ ▃ ▅ ▅ ▃ ▄ ▄ ▁
  ▂▃▃▇▁█▁█▁█▁█▁█▁█▁█▁█▁█▁█▁█▁█▁▇▁▇▁▆▁▅▁▅▁▄▁▄▁▄▁▃▁▃▁▂▁▂▁▂▁▂▁▂▂ ▄
  1.44 μs         Histogram: frequency by time        1.57 μs <

 Memory estimate: 1.66 KiB, allocs estimate: 60.

julia> @benchmark toidict($pairs)
BenchmarkTools.Trial: 10000 samples with 964 evaluations.
 Range (min … max):   90.076 ns …   1.652 μs  ┊ GC (min … max):  0.00% … 93.47%
 Time  (median):     102.524 ns               ┊ GC (median):     0.00%
 Time  (mean ± σ):   117.682 ns ± 152.728 ns  ┊ GC (mean ± σ):  13.49% ±  9.63%

  █▂                                                            ▁
  ██▄▃▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▅ █
  90.1 ns       Histogram: log(frequency) by time       1.57 μs <
```

Maybe the magnitude here doesn't matter that much in reality though...you'd have to have small arrow messages and lots of metadata, which doesn't seem like a likely/good case to optimize for...

so let's go with the simpler `foldl` as you suggest and can optimize later if it ever matters